lib/proxy-agent.js

```diff
@@ -55,8 +55,11 @@ class ProxyAgent extends DispatcherBase {
     this[kProxyTls] = opts.proxyTls
     this[kProxyHeaders] = {}
 
+    // @deprecated in favour of opts.token
     if (opts.auth) {
       this[kProxyHeaders]['proxy-authorization'] = `Basic ${opts.auth}`
+    } else if (opts.token) {
+      this[kProxyHeaders]['proxy-authorization'] = opts.token
```

Comment:
Maybe throw an error if both are provided?

Reply:
@ronag
I thought that would be more helpful too, so I fixed it.
https://github.com/nodejs/undici/pull/1705/commits/b989e96f3c8bc9f2bdc98d1152afcbe80c10daad